src/mavlink-router/mainloop.cpp

```diff
@@ -317,11 +317,15 @@ int Mainloop::run_single(int timeout_msec)
     }
     for (int i = 0; i < r; i++) {
         if (events[i].data.ptr == &_pipefd) {
-            _handle_pipe();
+            if (!(events[i].events & EPOLLERR)) {
```

Comment:
The error needs to be handled, otherwise there's a risk of busy-looping.
- `EPOLLHUP` should also be checked for, specifically for the pipe
- at the minimum, remove the fd, and print an error
- ideally try to reopen it

Reply:
The way this is handled for the other EPOLLERR is to exit and let the OS do the cleanup. Would that work?

Reply:
I'd rather reopen in the case of the pipe to avoid disrupting anything else (e.g. log streaming). It's likely that Nico was experiencing exactly that: a poll error.